app/renderer/components/main/siteInfo.js

```diff
@@ -3,259 +3,271 @@
 const platformUtil = require('../../../common/lib/platformUtil')
+const frameStateUtil = require('../../../../js/state/frameStateUtil')
 
 // Styles
 const globalStyles = require('../styles/global')
 const commonStyles = require('../styles/commonStyles')
 
-class SiteInfo extends ImmutableComponent {
-  constructor () {
-    super()
+class SiteInfo extends React.Component {
+  constructor (props) {
+    super(props)
     this.onAllowRunInsecureContent = this.onAllowRunInsecureContent.bind(this)
     this.onDenyRunInsecureContent = this.onDenyRunInsecureContent.bind(this)
     this.onViewCertificate = this.onViewCertificate.bind(this)
   }
+
   onAllowRunInsecureContent () {
-    appActions.changeSiteSetting(siteUtil.getOrigin(this.location),
-      'runInsecureContent', true, this.isPrivate)
-    appActions.loadURLRequested(this.props.frameProps.get('tabId'), this.location)
-    this.props.onHide()
+    appActions.changeSiteSetting(siteUtil.getOrigin(this.props.location),
+      'runInsecureContent', true, this.props.isPrivate)
+    appActions.loadURLRequested(this.props.activeTabId, this.props.location)
+    this.onHide()
   }
+
   onDenyRunInsecureContent () {
-    appActions.removeSiteSetting(siteUtil.getOrigin(this.location),
-      'runInsecureContent', this.isPrivate)
-    appActions.loadURLRequested(this.props.frameProps.get('tabId'), this.location)
-    this.props.onHide()
+    appActions.removeSiteSetting(siteUtil.getOrigin(this.props.location),
+      'runInsecureContent', this.props.isPrivate)
+    appActions.loadURLRequested(this.props.activeTabId, this.props.location)
+    this.onHide()
   }
+
   onViewCertificate () {
-    this.props.onHide()
+    this.onHide()
     webviewActions.showCertificate()
   }
-  get isExtendedValidation () {
-    return this.props.frameProps.getIn(['security', 'isExtendedValidation'])
-  }
-  get isSecure () {
-    return this.props.frameProps.getIn(['security', 'isSecure'])
-  }
-  get isPrivate () {
-    return this.props.frameProps.getIn(['isPrivate'])
-  }
-  get runInsecureContent () {
-    return this.props.frameProps.getIn(['security', 'runInsecureContent'])
-  }
-  get isBlockedRunInsecureContent () {
-    return this.props.frameProps.getIn(['security', 'blockedRunInsecureContent'])
-  }
-  get partitionNumber () {
-    return this.props.frameProps.getIn(['partitionNumber'])
-  }
-  get location () {
-    return this.props.frameProps.getIn(['location'])
-  }
-  get maybePhishingLocation () {
-    return isPotentialPhishingUrl(this.props.frameProps.getIn(['location']))
-  }
-  render () {
-    if (isSourceAboutUrl(this.location)) {
-      return null
-    }
 
-    // Figure out the partition info display
-    let l10nArgs = {
-      partitionNumber: this.partitionNumber
-    }
+  onHide () {
+    windowActions.setSiteInfoVisible(false)
+  }
 
-    let secureIcon
-    if (this.isSecure === true && !this.runInsecureContent) {
+  get secureIcon () {
```

Comment:
same as bellow. What we can test is if props are set correctly, so that we move for example `isFullySecured` into state helper and do a unit test there.